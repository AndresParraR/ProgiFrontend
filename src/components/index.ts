export * from './vehicleTotal';
export * from './general';
