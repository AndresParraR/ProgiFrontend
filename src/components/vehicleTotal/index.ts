export { default as TableVehicle } from './TableVehicle.vue';
export { default as FilterVehicle } from './FilterVehicle.vue';
export { default as DialogVehicle } from './DialogVehicle.vue';
