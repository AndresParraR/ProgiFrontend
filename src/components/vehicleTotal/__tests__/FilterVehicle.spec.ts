import { mount } from "@vue/test-utils";
import { FilterVehicle } from "@/components";
import { beforeEach, describe, expect, test } from "vitest";
import { Quasar } from "quasar";

describe("FilterVehicle Component", () => {
  let wrapper: any = null;

  beforeEach(() => {
    const global = {
      plugins: [Quasar],
    };

    wrapper = mount(FilterVehicle, {
      global,
    });
  });

  test("Should render correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  test("Should updateSearch emit searchUpdated", async () => {
    wrapper.vm.updateSearch("text");

    expect(wrapper.emitted().searchUpdated).toBeTruthy();
  });

  test("Should emit addVehicle", async () => {
    wrapper.vm.emit('addVehicle');

    expect(wrapper.emitted().addVehicle).toBeTruthy();
  });
});
