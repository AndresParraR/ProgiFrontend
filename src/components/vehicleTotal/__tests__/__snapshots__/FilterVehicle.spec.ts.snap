// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FilterVehicle Component > Should render correctly 1`] = `
<div
  class="q-card"
>
  <div
    class="q-card__section q-card__section--vert"
  >
    <div
      class="col-xs-12 row q-col-gutter-md"
    >
      <div
        class="q-space"
      />
      <div
        class="col-xs-12 col-sm-4 col-md-1"
      >
        <button
          class="q-btn q-btn-item non-selectable no-outline q-btn--push q-btn--round bg-primary text-white q-btn--actionable q-focusable q-hoverable"
          tabindex="0"
          type="button"
        >
          <span
            class="q-focus-helper"
          />
          <span
            class="q-btn__content text-center col items-center q-anchor--skip justify-center row"
          >
            <i
              aria-hidden="true"
              class="q-icon notranslate material-icons"
              role="img"
            >
              add
            </i>
          </span>
        </button>
      </div>
      <div
        class="col-xs-12 col-sm-6 col-md-6"
      >
        <label
          class="q-field row no-wrap items-start q-field--outlined q-input q-field--rounded filter-outlined"
          for="f_f2b4c347-93cc-429d-8d02-2ef007218ad6"
        >
          <!---->
          <div
            class="q-field__inner relative-position col self-stretch"
          >
            <div
              class="q-field__control relative-position row no-wrap"
              tabindex="-1"
            >
              <div
                class="q-field__control-container col relative-position row no-wrap q-anchor--skip"
              >
                <input
                  class="q-field__native q-placeholder"
                  id="f_f2b4c347-93cc-429d-8d02-2ef007218ad6"
                  placeholder="Buscar"
                  tabindex="0"
                  type="text"
                />
                <!---->
              </div>
              <div
                class="q-field__append q-field__marginal row no-wrap items-center"
              >
                <i
                  aria-hidden="true"
                  class="q-icon text-primary notranslate material-icons"
                  role="presentation"
                >
                  search
                </i>
              </div>
            </div>
            <!---->
          </div>
          <!---->
        </label>
      </div>
    </div>
  </div>
</div>
`;
